src/frontend/js/pages/TeacherDashboardContractsLayout/hooks/useDownloadContractArchive/index.tsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from 'react';
+import useContractArchive from 'hooks/useContractArchive';
+import { useContracts } from 'hooks/useContracts';
+import { ContractState } from 'types/Joanie';
+import { PER_PAGE } from 'settings';
+import useCheckContractArchiveExists from '../useCheckContractArchiveExists';
+import {
+  storeContractArchiveId,
+  unstoreContractArchiveId,
+  getStoredContractArchiveId,
+  isStoredContractArchiveIdExpired,
+} from './contractArchiveLocalStorage';
+
+export enum ContractDownloadStatus {
+  IDDLE = 'iddle',
+  PENDING = 'pending',
+  READY = 'ready',
+}
+
+interface UseTeacherContractsBulkDownloadProps {
+  organizationId: string;
+}
+
+const useDownloadContractArchive = ({ organizationId }: UseTeacherContractsBulkDownloadProps) => {
+  const {
+    items: contracts,
+    states: { isFetched },
+  } = useContracts({
+    organization_id: organizationId,
+    signature_state: ContractState.SIGNED,
+    page: 1,
```

Comment:
I do not have digged about this but do you think that only working on page 1 of contracts could cause any bugs   in this hook?